### ISSUE TITLE
Truffle Errors When Downloading Using Windows Powershell

### ISSUE THREAD

**Issue by TechGuru29** [trufflesuite/truffle#4868]
npm ERR! code 1
npm ERR! path C:\Users\KingBrad\AppData\Roaming\npm\node_modules\truffle\node_modules\ganache\node_modules\leveldown
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node-gyp rebuild
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@5.1.0
npm ERR! gyp info using node@14.15.1 | win32 | x64
npm ERR! gyp info find Python using Python version 3.10.2 found at "C:\Users\KingBrad\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\python.exe"
npm ERR! gyp ERR! find VS
npm ERR! gyp ERR! find VS msvs_version not set from command line or npm config
npm ERR! gyp ERR! find VS VCINSTALLDIR not set, not running in VS Command Prompt
npm ERR! gyp ERR! find VS checking VS2019 (16.11.32228.343) found at:
npm ERR! gyp ERR! find VS "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community"
npm ERR! gyp ERR! find VS - found "Visual Studio C++ core features"
npm ERR! gyp ERR! find VS - missing any VC++ toolset
npm ERR! gyp ERR! find VS could not find a version of Visual Studio 2017 or newer to use
npm ERR! gyp ERR! find VS looking for Visual Studio 2015
npm ERR! gyp ERR! find VS - not found
npm ERR! gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
npm ERR! gyp ERR! find VS
npm ERR! gyp ERR! find VS **************************************************************
npm ERR! gyp ERR! find VS You need to install the latest version of Visual Studio
npm ERR! gyp ERR! find VS including the "Desktop development with C++" workload.
npm ERR! gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
npm ERR! gyp ERR! find VS **************************************************************
npm ERR! gyp ERR! find VS
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: Could not find any Visual Studio installation to use
npm ERR! gyp ERR! stack     at VisualStudioFinder.fail (C:\Users\KingBrad\node_modules\node-gyp\lib\find-visualstudio.js:121:47)
npm ERR! gyp ERR! stack     at C:\Users\KingBrad\node_modules\node-gyp\lib\find-visualstudio.js:74:16
npm ERR! gyp ERR! stack     at VisualStudioFinder.findVisualStudio2013 (C:\Users\KingBrad\node_modules\node-gyp\lib\find-visualstudio.js:351:14)
npm ERR! gyp ERR! stack     at C:\Users\KingBrad\node_modules\node-gyp\lib\find-visualstudio.js:70:14
npm ERR! gyp ERR! stack     at C:\Users\KingBrad\node_modules\node-gyp\lib\find-visualstudio.js:372:16
npm ERR! gyp ERR! stack     at C:\Users\KingBrad\node_modules\node-gyp\lib\util.js:54:7
npm ERR! gyp ERR! stack     at C:\Users\KingBrad\node_modules\node-gyp\lib\util.js:33:16
npm ERR! gyp ERR! stack     at ChildProcess.exithandler (child_process.js:315:5)
npm ERR! gyp ERR! stack     at ChildProcess.emit (events.js:315:20)
npm ERR! gyp ERR! stack     at maybeClose (internal/child_process.js:1048:16)
npm ERR! gyp ERR! System Windows_NT 10.0.19044
npm ERR! gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\KingBrad\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd C:\Users\KingBrad\AppData\Roaming\npm\node_modules\truffle\node_modules\ganache\node_modules\leveldown
npm ERR! gyp ERR! node -v v14.15.1
npm ERR! gyp ERR! node-gyp -v v5.1.0
npm ERR! gyp ERR! not ok

---------------------------

## Issue

What the issue is, in broad strokes.

Please provide the shortest amount of steps to reproduce your issue. We tried using this command before running the install of truffle to see if it would fix the issue still get all of the ERRORS.  set-ExecutionPolicy -scope process -ExecutionPolicy Bypass

## Expected Behavior

What you expected to happen. I expect for truffle to install on my PC when I run npm install -g truffle

## Actual Results

What actually happened. Please give examples and support it with screenshots, copied output or error messages.

## Environment

* Operating System: Windows 10 Home 
* Ethereum client: Ganache 
* Truffle version (`truffle version`): None yet will not install.
* node version (`node --version`): 14.15.1
* npm version (`npm --version`): 8.5.4

**Comment by Tab007** [comment]
I have the same problem

**Comment by lsqproduction** [comment]
What command did you use to install?

**Comment by lawweiliang** [comment]
Hi this is what I do.

If you insist of using Truffle like me.
* Install node 12.22.9
* Running different version of truffle without any problem

If you insist using Latest version of node 16.14.1. npm 8.5.0
* My suggestion is using HardHat rather than Truffle
* If you install truffle, you are going to face a lot of problems. Even after you switched the npm to version 6. You are still facing a lot of problems.
* If I am the team lead, I will never spend my time(company resource) to solve truffle incompatible problem. 
   * Alway remember, we use the tool because it save our time. If it is not, we go for a better framework/tools. 
 Hope its help. :)

**Comment by cds-amal** [comment]
@TechGuru29, @Tab007, @lawweiliang, did you follow the instructions that `npm` provided in the error message? This is an indication that your NodeJS environment misconfigured, specifically missing the [build tools dependencies](https://nodejs.org/en/docs/meta/topics/dependencies/#dependencies). 

>npm ERR! gyp ERR! find VS **************************************************************
npm ERR! gyp ERR! find VS You need to install the latest version of Visual Studio
npm ERR! gyp ERR! find VS including the "Desktop development with C++" workload.
npm ERR! gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
npm ERR! gyp ERR! find VS **************************************************************

**Comment by ezeamaka2** [comment]
I had similar issues when I was installing Truffle. But, I was able to solve the issue. For anyone having similar problem, here is what I did....

 - Downland and install the latest version of nodejs
 - Downland and install python 3.
 
When installing the python 3, check the option, **Add path to environmental variable**.

If you installed python 3 without checking that option. You will need to add it manually to the environmental variable. 

After the installations has been successfully completed. Restart the PC and try install truffle again with 
**npm install -g truffle**

I hope this works for you.
